File: JPAPractice/src/main/java/com/example/jpa_project/ch06Example/Member.java
package com.example.jpa_project.ch06Example;

import lombok.Data;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
public class Member {
    @Id
    @GeneratedValue
    @Column(name = "MEMBER_ID")
    private Long id;

    private String username;

    @OneToMany(mappedBy = "member")
    private List<MemberProduct> memberProducts;

//    @OneToOne
//    @JoinColumn(name = "LOCKER_ID")
//    private Locker locker;

//    @ManyToOne
//    @JoinColumn(name = "TEAM_ID", insertable = false, updatable = false)
//    private Team team;

//    @ManyToMany
//    @JoinTable(
//            name = "MEMBER_PRODUCT",
//            joinColumns = @JoinColumn(name = "MEMBER_ID"),
//            inverseJoinColumns = @JoinColumn(name = "PRODUCT_ID")
//    )
//    private List<Product> products = new ArrayList<>();
}

File: kd_kotlin/src/main/kotlin/entity/Member.kt
package entity

import lombok.Data
import javax.persistence.Column
import javax.persistence.Embedded
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.OneToMany

@Entity
@Data
class Member (
    @Id @GeneratedValue @Column(name = "MEMBER_ID")
    var id: Long? = null,

    var name: String,

    @Embedded
    var address: Address,

    @OneToMany(mappedBy = "member")
    var orders: List<Order> ?= null
        )